src/include/duckdb/main/appender.hpp

```diff
@@ -116,12 +116,17 @@ class Appender : public BaseAppender {
 	shared_ptr<ClientContext> context;
 	//! The table description (including column names)
 	unique_ptr<TableDescription> description;
+	//! The default expressions
+	vector<optional_ptr<const ParsedExpression>> defaults;
```

Comment:
Perhaps we can store the bound expressions here instead?

We should also always be able to have a default value - if there is none the default value is `NULL`.

Reply:
If we bind at creation of the Appender, don't we run the risk of the binding being stale during inserts?

Reply:
Stale as in the defaults might change? I would say we don’t need to support changing defaults while the appender is running.